### PR TITLE
Update React 17

### PR DESCRIPTION
**Pull request by liywjl**
## What does this change?

Update react to v17

## Why?

We want to support JSX

## Screenshots

Tested on the platforms by forcing the banner with `#force-braze-message`.

Frontend:

Screenshot 2021-05-13 at 11 47 16

DCR:

Screenshot 2021-05-13 at 12 13 54

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -70,7 +70,8 @@
     "np": "^7.2.0",
     "preact-x": "npm:preact@10.3.1",
     "prettier": "^2.0.5",
-    "react": "16.13.x",
+    "react": "17.0.2",
+    "react-dom": "^17.0.2",
```
Comment (liywjl) [inline package.json:74]: `react-dom` is needed in order to run unit test somehow? unsure why this is needed

Comment (tjmw) [reply package.json:76]: Yep I think it's needed by react-testing-library IIRC.

**Comment by tjmw** [comment]
Hey @liywjl thanks for opening the PR. Sorry it's taken me a while to get to this! The React 17 upgrade is part of #123 but no sense in coupling them together so I'll plan on merging this PR soon. I checked out this branch locally and found a few steps I needed to take to get Storybook and the build working correctly. I've done a quick integration test on DCR and all seems well, I still need to do the same for frontend but I think that's the only outstanding task.

**Review by michaelclapham — APPROVED** [review]
Builds locally and code looks good to me!

**Pull request merged**